src/Renci.SshNet/PrivateKeyAuthenticationMethod.cs

```diff
@@ -111,7 +111,7 @@ public override AuthenticationResult Authenticate(Session session)
 
                     session.WaitOnHandle(_authenticationCompleted);
 
-                    if (_authenticationResult == AuthenticationResult.Success)
+                    if (_authenticationResult is AuthenticationResult.Success or AuthenticationResult.PartialSuccess)
```

Comment:
I *think* this change makes sense. Was there something not working which motivated the change?

Reply:
You can remove this line locally which will cause one test to fail. This happened when I removed PubkeyAcceptedAlgorithms from the config. I think this change is okay, but I'd love it if you checked out this line yourself

Reply:
I am not super familiar with multi-factor auth in SSH, but the reason for the failure and the fix both make sense to me. Thanks